### ISSUE TITLE
Vue VList reactivity not working

### ISSUE THREAD

**Issue by k1nz** [inokawa/virtua#557]
**Describe the bug**
The VList data reactivity not working in virtua/vue
**To Reproduce**

Click the update button and the list data is not updated

https://codesandbox.io/p/sandbox/virtua-vue-reactivity-bug-qs5n56?file=%2Fsrc%2FApp.vue%3A1%2C1&workspaceId=46b18128-5751-4287-b6a6-40706a4c08fe

```html
<script setup>
import { VList } from 'virtua/vue'
import { ref } from 'vue'

const sizes = [20, 40, 180, 77]
const data = ref(Array.from({ length: 1000 }).map((_, i) => sizes[i % 4]))
const updateData = () => {
  data.value = []
}
</script>

<template>
  <button @click="updateData">update</button>
  <VList :data="data" :style="{ height: '800px' }" #default="{ item, index }">
    <div
      :key="index"
      :style="{
        height: item + 'px',
        background: 'white',
        borderBottom: 'solid 1px #ccc',
      }"
    >
      {{ index }}
    </div>
  </VList>
</template>
```

**Platform:**

- OS: MacOS
- Browser: Chrome
- Version of this package: 0.37.4
- Version of framework: vue@3.2.45

**Comment by inokawa** [comment]
Duplicate of #552

**Comment by inokawa** [comment]
Fixed in `0.38.5`.
